### PR TITLE
Add more deep handling for P2C dates

### PR DESCRIPTION
**Pull request by fabricantes**

### CODE REVIEW

File: pepdb/tasks/management/commands/apply_smida_report.py [hunk 1]
```diff
@@ -153,7 +154,29 @@ def handle(self, *args, **options):
...
+                        if options["real_run"]:
+                            prev_position.save()
+
+                        tqdm.write("Updated previous position for SMIDACandidate ID: {}"
+                                   .format(candidate.id))
+
+                else:
+                    date_established = candidate.dt_of_first_entry
+
+                date_finished = self.p2c_get_date_finished(candidate) or candidate.dt_of_last_entry
```
Comment (dchaplinsky) [inline pepdb/tasks/management/commands/apply_smida_report.py]: You should account for the fact, that one might be listed in an quarter report, that has end date 31.12.2018, but that doesn't mean, that connection is over. Connection is only over if the end date of the last report is in the past (more than a quarter ago)

Comment (fabricantes) [reply pepdb/tasks/management/commands/apply_smida_report.py]: Makes sense. Added a fix

File: pepdb/tasks/management/commands/apply_smida_report.py [hunk 2]
```diff
@@ -444,6 +451,15 @@ def new_persons_having_multiple_company_relations(self):
             if person.companies_cnt > 1:
                 tqdm.write("{} {}".format(person.url_uk, person.companies_cnt))
 
+    def threshold_quarter_end(self):
+        """
+        Calculate date behind which the reports would considered as the past
+        :return: date
+        """
+        th_date = datetime.now() - relativedelta(months=6)
+        th_quarter = (th_date.month-1) / 3 + 1
```
Comment (dchaplinsky) [inline pepdb/tasks/management/commands/apply_smida_report.py]: Please make it return int explicitly, do not rely on python2.7 flaw in division.

Also, spaces around -

Comment (fabricantes) [reply pepdb/tasks/management/commands/apply_smida_report.py]: Done

**Pull request merged**